src/python/CRABClient/UserUtilities.py

```diff
@@ -120,46 +120,56 @@ def getFileFromURL(url, filename = None, proxyfilename = None):
     """
-    parsedurl = urlparse(url)
+    parsedUrl = urlparse(url)
     if filename == None:
-        path = parsedurl.path
+        path = parsedUrl.path
         filename = os.path.basename(path)
-    try:
-        opener = urllib.URLopener(key_file = proxyfilename, cert_file = proxyfilename)
-        socket = opener.open(url)
-        status = socket.getcode()
-        # Read the file by chunks instead of all at once, appending each chunk to the final result.
-        # This lowers the memory overhead, which can be a problem with big files.
-        with open (filename, 'a') as f:
-            f.seek(0)
-            f.truncate()
-            while True:
-                piece = socket.read(1024)
-                if not piece:
-                    break
-                f.write(piece)
-    except IOError as ioex:
-        msg = "Error while trying to retrieve file from %s: %s" % (url, ioex)
-        msg += "\nMake sure the URL is correct."
-        exc = ClientException(msg)
-        if ioex[0] == 'http error':
-            exc.status = ioex[1]
-        raise exc
-    except Exception as ex:
-        tblogger = logging.getLogger('CRAB3')
-        tblogger.exception(ex)
-        msg = "Unexpected error while trying to retrieve file from %s: %s" % (url, ex)
-        raise ClientException(msg)
-    if status != 200 and parsedurl.scheme in ['http', 'https']:
-        exc = ClientException("Unable to retieve the file from %s. HTTP status code %s. HTTP content: %s" % (url, status, socket.info()))
-        exc.status = status
-        raise exc
+
+    data = getDataFromURL(url, proxyfilename)
+
+    if data:
+        try:
+            with open(filename, 'a') as f:
+                f.seek(0)
+                f.truncate()
+                f.write(data)
+        except IOError as ex:
+            logger = logging.getLogger('CRAB3')
+            logger.exception(ex)
+            msg = "Error while writing %s. Got:\n%s" \
+                    % (filename, ex)
+            raise ClientException(msg)
+
     return filename
 
 
+def getDataFromURL(url, proxyfilename = None):
+    """
+    Read the content of a URL and return it as a string.
+    Type of content should not matter, it can be a json file or a tarball for example.
+
+    url: the link you would like to retrieve
+    proxyfilename: the x509 proxy certificate to be used in case auth is required
+
+    Returns binary data encoded as a string, which can be later processed
+    according to what kind of content it represents.
+    """
+
+    from httplib import HTTPException
+    #CRABServer dependency
+    from RESTInteractions import HTTPRequests
+
+    reqHandler = RequestHandler()
+    try:
+        _, data = reqHandler.request(url=url, params={}, ckey=proxyfilename,
+                cert=proxyfilename, capath=HTTPRequests.getCACertPath())
+    except HTTPException as ex:
```

Comment:
Printing the exception should always give the headers, example with my current implementation [1]. Also I'd rather be able to decide case-by-case whether the printout to stdout/log is needed and handle it in the code depending on this method.

[1]
```
[erupeika@lxplus036 crab_projects]$ crab report
Running crab status first to fetch necessary information.
Error while getting status information. Got:
Cannot retrieve the status_cache file. Maybe the task process has not run yet? Got:
url=https://cmsweb-testbed.cern.ch/scheddmon/0196/cms1425/170515_115428:erupeika_crab_myproxy0/status_cachesss, code=404, reason=Not Found, headers={'Content-Length': '346', 'Set-Cookie': 'cms-auth=61c6ab89eda7a74c7f55a10fdb528680f89e4ecf90f2b1841597a7d5275a643ecc669cbf8da581a1;path=/;secure;httponly;expires=Thu, 01-Jan-1970 00:00:01 GMT', 'Server': 'Apache/2.2.15 (Red Hat)', 'Date': 'Mon, 15 May 2017 14:38:48 GMT', 'Content-Type': 'text/html; charset=iso-8859-1', 'CMS-Server-Time': 'D=4057 t=1494859128101405'} 
Error: Status information is unavailable, will not proceed with the report. Try again a few minutes later if the task has just been submitted.
Log file is /afs/cern.ch/user/e/erupeika/crab_projects/crab_myproxy0/crab.log
```